src/libraries/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -20,42 +21,57 @@ internal static partial class Sys
         /// Returns the number of bytes placed into the buffer on success; bufferSize if the buffer is too small; and -1 on error.
         /// </returns>
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_ReadLink", SetLastError = true)]
-        private static extern int ReadLink(string path, byte[] buffer, int bufferSize);
+        private static extern int ReadLink(ref byte path, byte[] buffer, int bufferSize);
 
         /// <summary>
         /// Takes a path to a symbolic link and returns the link target path.
         /// </summary>
-        /// <param name="path">The path to the symlink</param>
-        /// <returns>
-        /// Returns the link to the target path on success; and null otherwise.
-        /// </returns>
-        public static string? ReadLink(string path)
+        /// <param name="path">The path to the symlink.</param>
+        /// <returns>Returns the link to the target path on success; and null otherwise.</returns>
+        internal static string? ReadLink(ReadOnlySpan<char> path)
         {
-            int bufferSize = 256;
-            while (true)
+            int outputBufferSize = DefaultPathBufferSize;
+
+            // Use an initial buffer size that prevents disposing and renting
+            // a second time when calling ConvertAndTerminateString.
+            int pathBufferSize = Encoding.UTF8.GetMaxByteCount(path.Length) + 1;
+            var converter = new ValueUtf8Converter(stackalloc byte[pathBufferSize]);
```

Comment:
Remove the try that disposes this instance.
```suggestion
            using var converter = new ValueUtf8Converter(stackalloc byte[pathBufferSize]);
```

Reply:
I think I never asked, why `ValueUtf8Converter` needs to be used?